drivers/SmartThings/bose/src/init.lua

```diff
@@ -300,6 +300,28 @@ local bose = Driver("bose", {
   },
 })
 
+local function ip_change_check()
+  local id_to_device = {}
+  local device_list = bose:get_devices()
+  for _, device in ipairs(device_list) do
+    local id = bose_utils.get_serial_number(device)
+    id_to_device[id] = device
+  end
+  discovery.find(nil, function(found)
+    local known = id_to_device[found.id]
+    if known ~= nil then
+      local known_ip = known:get_field("ip")
+      if known_ip == nil or known_ip ~= found.ip then
+        log.trace("Updating device ip:", found.id, found.ip)
```

Comment:
I think this should be info level and in hub logs, since it represents a moment when we essentially re-discover the device on the network.